v1/common/amqp.go

```diff
@@ -4,23 +4,47 @@ import (
 	"crypto/tls"
 	"fmt"
 	"sync"
+	"time"
 
 	"github.com/streadway/amqp"
 )
 
 // AMQPConnector ...
 type AMQPConnector struct {
 	connManager *amqpConnectionManager
+
+	exchangeMaxRetries   int
+	exchangeRetryTimeout time.Duration
 }
 
 func NewAMQPConnector(url string, tlsConfig *tls.Config) *AMQPConnector {
 	return &AMQPConnector{
 		connManager: newAMQPConnectionManager(url, tlsConfig),
+
+		exchangeMaxRetries:   3,
+		exchangeRetryTimeout: time.Second,
 	}
 }
 
 // Exchange declares an exchange, opens a channel declares and binds the queue and enables publish notifications using the existing RabbitMQ connection.
 func (ac *AMQPConnector) Exchange(exchange, exchangeType, queueName string, queueDurable, queueDelete bool, queueBindingKey string, exchangeDeclareArgs, queueDeclareArgs, queueBindingArgs amqp.Table) (*amqp.Channel, amqp.Queue, <-chan amqp.Confirmation, error) {
+	var lastErr error
+	for retry := 0; retry < ac.exchangeMaxRetries; retry++ {
+		channel, queue, confirmChan, err := ac.exchange(exchange, exchangeType, queueName, queueDurable, queueDelete, queueBindingKey, exchangeDeclareArgs, queueDeclareArgs, queueBindingArgs)
+		if err != nil {
+			lastErr = err
+			_, connErr := err.(amqpConnError)
```

Comment:
I don't think this works, because the error message and it's type are changed by the this line below:
```go
return nil, amqp.Queue{}, nil, fmt.Errorf("Exchange declare error: %s", err)
```

Can you test that this works by manually returning this connection error from `channel.ExchangeDeclare`?